frontend/src/hocs/withScreencast/withScreencast.tsx

```diff
@@ -0,0 +1,267 @@
+/**
+ * @license
+ * Copyright 2018-2020 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from "classnames"
+import WhichBrowser from "which-browser"
+import ScreenCastRecorder from "lib/ScreenCastRecorder"
+import hoistNonReactStatics from "hoist-non-react-statics"
+import React, { PureComponent, ComponentType, ReactNode } from "react"
+
+import {
+  UnsupportedBrowserDialog,
+  ScreencastDialog,
+  VideoRecordedDialog,
+} from "hocs/withScreencast/components"
+
+import "./withScreencast.scss"
+
+interface WithScreenCastProps {}
+
+interface WithScreenCastState {
+  fileName: string
+  recording: boolean
+  recordAudio: boolean
+  countdown: number
+  startAnimation: boolean
+  blob?: Blob
+  showRecordedDialog: boolean
+  showScreencastDialog: boolean
+  showUnsupportedDialog: boolean
+}
+
+export interface ScreenCastHOC extends WithScreenCastState {
+  toggleRecordAudio: () => void
+  startRecording: (fileName: string) => void
+  stopRecording: () => void
+}
+
+const SUPPORTED_BROWSERS: any = {
+  Chrome: "49",
```

Comment:
These numbers don't match the ones on this page: https://caniuse.com/#search=getdisplaymedia